var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -24,8 +24,11 @@
     "gfx906": "Mi50",
     "gfx908": "Mi100",
     "gfx90a": "Mi250",
+    "gfx942": "Mi300",
     "gfx90a:xnack-": "Mi250",
     "gfx90a:xnack+": "Mi250",
+    "gfx942:xnack-": "Mi300",
+    "gfx942:xnack+": "Mi300",
```

Comment:
The list of supported targets in https://github.com/spack/spack/blob/9265991767e6fbed0e8de4bdcf49506a7b9a08ed/lib/spack/spack/build_systems/rocm.py#L95-L131 does not include `gfx942:xnack-`/`+`, but I don't know if this is simply missing there, or if it hurts to have them here even if they can't be used.

@dtaller I see you listed as maintainer for the rocm build system: should all the architectures have the xnack variants, or does it really only apply to some?

Regardless of the above, I suppose this could be refactored to only include the `gfx942` part and then index into the map using `target.split(":")[0]` or something in that direction, but I'm not requesting that as part of this PR since `gfx90a` is already there with `xnack`.